### PR TITLE
Support table comments on create table in Kudu 

### PR DESCRIPTION
**Pull request by jon-esperanza**
## Description
This PR adds support for trino-kudu to create table with table comment using the Java Kudu Client.
This change allows users to use Trino to create tables with table comments when using our Apache Kudu connector.

## Additional context and related issues
Fixes the table comment part of #12469 , another PR will be made for create table with column comments.

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Kudu
* Add support for table comments when creating tables. ({issue}`17945`)
```

### CODE REVIEW

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java [hunk 1]
```diff
@@ -161,14 +161,16 @@ private ConnectorTableMetadata getTableMetadata(KuduTableHandle tableHandle)
     {
         KuduTable table = tableHandle.getTable(clientSession);
         Schema schema = table.getSchema();
+        // Kudu table comments are not null, instead empty strings
+        Optional<String> tableComment = (table.getComment().isEmpty()) ? Optional.empty() : Optional.ofNullable(table.getComment());
```
Comment (jon-esperanza) [inline plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java:165]: This is just to conform the way Kudu handles null comments to the way Trino handles null comments.

Kudu: `''` => Trino: `null`

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/BaseKuduConnectorSmokeTest.java [hunk 2]
```diff
@@ -150,4 +150,20 @@ public void testUpdate()
...
+
+    @Test
+    public void testCreateTableWithTableComment()
+    {
+        String tableName = "test_create_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a bigint WITH (primary_key=true)) COMMENT 'test comment' WITH (partition_by_hash_columns = ARRAY['a'], partition_by_hash_buckets = 2)");
+
+        if (getKuduServerVersion().equals(EARLIEST_TAG)) {
+            assertNull(getTableComment(tableName));
```
Comment (jon-esperanza) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/BaseKuduConnectorSmokeTest.java:162]: Could not find a way to throw an exception when user attempts to add table comment on EARLIEST_TAG. If we don't want to fail silently, we will need to figure out a way to check kudu version at runtime?

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java (hunk 1 above)
Comment (jon-esperanza) [reply plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java:165]: This is a pretty trivial expression.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java [hunk 3]
```diff
@@ -418,8 +417,8 @@ public void testCreateTable()
         final String finalTableName = tableName;
         assertThatThrownBy(() -> assertUpdate(
                 "CREATE TABLE " + tableNameLike + " (LIKE " + finalTableName + ", " +
-                    "d bigint, e varchar(50))" +
-                    "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)"))
+                        "d bigint, e varchar(50))" +
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:420]: Please separate a commit. This change is unrelated to table comment support.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java [hunk 4]
```diff
@@ -990,6 +989,27 @@ public void testUpdateRowConcurrently()
         throw new SkipException("Kudu doesn't support concurrent update of different columns in a row");
     }
 
+    @Test
+    @Override
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:993]: Please leave a comment. Same for others.
```
        // TODO Remove this overriding test once kudu connector can create tables with default partitions
```

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java (hunk 1 above)
Comment (ebyhr) [inline plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java:165]: No need to use a ternary operator. 
```java
Optional.ofNullable(emptyToNull(table.getComment()));
```

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java [hunk 5]
```diff
@@ -990,6 +989,27 @@ public void testUpdateRowConcurrently()
...
+    public void testCreateTableWithTableComment()
+    {
+        String tableName = "test_create_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a bigint WITH (primary_key=true)) COMMENT 'test comment' WITH (partition_by_hash_columns = ARRAY['a'], partition_by_hash_buckets = 2)");
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:998]: This line is too long. Please wrap after `COMMENT 'test comment'`.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java [hunk 6]
```diff
@@ -990,6 +989,27 @@ public void testUpdateRowConcurrently()
...
+        assertEquals(getTableComment(getSession().getCatalog().orElseThrow(), getSession().getSchema().orElseThrow(), tableName), "test comment");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "testCommentDataProvider")
+    @Override
+    public void testCreateTableWithTableCommentSpecialCharacter(String comment)
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_create_", "(a bigint WITH (primary_key=true)) COMMENT " + varcharLiteral(comment) + "WITH (partition_by_hash_columns = ARRAY['a'], partition_by_hash_buckets = 2)")) {
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:1008]: This line is too long. Please wrap per argument.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java [hunk 7]
```diff
@@ -990,6 +989,27 @@ public void testUpdateRowConcurrently()
...
+            assertEquals(getTableComment(getSession().getCatalog().orElseThrow(), getSession().getSchema().orElseThrow(), table.getName()), comment);
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:1009]: nit: We don't need to get catalog and schema from session:
```java
assertEquals(getTableComment("kudu", "default", table.getName()), comment);
```

**Comment by ebyhr** [comment]
@jon-esperanza Did you already submit CLA?

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java (hunk 1 above)
Comment (jon-esperanza) [reply plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduMetadata.java:165]: nice!

**Comment by jon-esperanza** [comment]
> @jon-esperanza Did you already submit CLA? 

Yes, submitted under jonathanesperanza67@gmail.com

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java (hunk 3 above)
Comment (jon-esperanza) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:420]: my ide formatter did this, not sure why. will revert

Comment (jon-esperanza) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:420]: made new commit for reformatting this file

**Comment by ebyhr** [comment]
Let me re-review once your CLA is registered.

**Comment by jon-esperanza** [comment]
> Let me re-review once your CLA is registered. 

I got email confirmation that the CLA was registered.

**Comment by ebyhr** [comment]
@cla-bot check

**Review by ebyhr — APPROVED** [review]
